mobilesdk/zboxapi/client.go

```diff
@@ -18,12 +18,29 @@ import (
 
 var (
 	zboxApiClient            *zboxapi.Client
-	ErrZboxApiNotInitialized = errors.New("0box: zboxapi client is not initialized")
+	ErrZboxApiNotInitialized = errors.New("zboxapi: zboxapi client is not initialized")
+	ErrZboxApiInvalidWallet  = errors.New("zboxapi: invalid wallet")
 )
 
 func Init(baseUrl, appType string) {
 	zboxApiClient = zboxapi.NewClient(baseUrl, appType)
-	zboxApiClient.SetWallet(client.GetClientID(), client.GetClientPrivateKey(), client.GetClientPublicKey())
+
+	c := client.GetClient()
+	if c != nil {
+		SetWallet(client.GetClientID(), client.GetClientPrivateKey(), client.GetClientPublicKey()) //nolint: errcheck
+	}
+}
+
+func SetWallet(clientID, clientPrivateKey, clientPublicKey string) error {
+	if zboxApiClient == nil {
+		return ErrZboxApiNotInitialized
+	}
+	if clientID != "" && clientPrivateKey != "" && clientPublicKey != "" {
+		zboxApiClient.SetWallet(clientID, clientPrivateKey, clientPublicKey)
+		return nil
+	}
+
+	return nil
```

Comment:
Shouldn't we return error if any of the client id or keys are empty?

Reply:
fixed